airbyte-cdk/python/airbyte_cdk/sources/file_based/config/file_based_stream_config.py

```diff
@@ -6,9 +6,9 @@
 from enum import Enum
 from typing import Any, Dict, List, Mapping, Optional, Union
 
-from pydantic import BaseModel, validator
+from pydantic import BaseModel, Field, validator
 
-PrimaryKeyType = Optional[Union[str, List[str], List[List[str]]]]
+PrimaryKeyType = Optional[Union[str, List[str]]]
```

Comment:
As per Sherif's comment in the other thread, ideally we'd want just `List[str]`. Looking over the existing s3 spec, I don't see any concept of a primary key, but it might be safer to allow for str as well since that is a pretty common way they are specified for other sources